source/Missions/Services/Agents/Handlers/AgentDamageHandler.cs

```diff
@@ -68,13 +71,43 @@ private void AgentDamageSend(MessagePayload<AgentDamaged> payload)
                 return;
             };
 
-            NetworkAgentDamaged message = new NetworkAgentDamaged(
+            // Handles friend fire event
+            if (networkAgentRegistry.IsControlled(victimId))
+            {
+                NetworkAgentDamaged friendlyFireMessage = new NetworkAgentDamaged(
+                    attackerId,
+                    victimId,
+                    payload.What.AttackCollisionData,
+                    payload.What.Blow);
+                networkMessageBroker.PublishNetworkEvent(friendlyFireMessage);
+                return;
+            }
+
+                if (networkAgentRegistry.TryGetExternalController(payload.What.VictimAgent, out NetPeer netPeer) == false) return;
+
+            NetworkDamageAgent message = new NetworkDamageAgent(
                 attackerId, 
                 victimId, 
                 payload.What.AttackCollisionData, 
                 payload.What.Blow);
 
-            networkMessageBroker.PublishNetworkEvent(message);
+            networkMessageBroker.PublishNetworkEvent(netPeer, message);
+        }
+
+        private void AgentDamageCheck(MessagePayload<NetworkDamageAgent> payload)
+        {
+            if (networkAgentRegistry.TryGetAgent(payload.What.VictimAgentId, out Agent resolvedAgent) == false) return;
+
+            if (resolvedAgent.Health <= 0) return;
+
+            NetworkAgentDamaged message = new NetworkAgentDamaged(
+                payload.What.AttackerAgentId,
+                payload.What.VictimAgentId,
+                payload.What.AttackCollisionData,
+                payload.What.Blow);
+
+            networkMessageBroker.PublishNetworkEventExcept((NetPeer)payload.Who, message);
+            AgentDamagePatch.OverrideAgentDamage(resolvedAgent, payload.What.Blow, payload.What.AttackCollisionData);
```

Comment:
https://github.com/Bannerlord-Coop-Team/BannerlordCoop/pull/569/files#diff-287e46b349c35d1040c626fe6504174f6fed59cb89ec8f0dd8683cb65fec343aR168 should use the same line here, also the prefix should skip for every case. Let me know if that isn't straight forward